### ISSUE TITLE
Rails/HasManyOrHasOneDependent flagging has_many: associations with foreign_key on_delete: cascade

### ISSUE THREAD

**Issue by robbl-as** [bbatsov/rubocop#5109]
Rails/HasManyOrHasOneDependent seems to be wrongly flagging has_many associations that have a foreign key with options `{on_delete: :cascade}`. We prefer to have the database to delete associated VerifyReminders, when we destroy a Signature and not Rails.

I have the following model:

```ruby
class Signature < ApplicationRecord
  has_many :verify_reminders
end
```

I have the following migration:

```ruby
class CreateVerifyReminders < ActiveRecord::Migration[5.1]
  def change
    create_table :verify_reminders, id: :uuid, default: -> { 'uuid_generate_v4()' } do |t|
      t.references :signature, type: :uuid, foreign_key: {on_delete: :cascade}
    end
  end
end
```

I have the following schema.rb after migration:

```ruby
ActiveRecord::Schema.define(version: 20171119105158) do
  create_table "verify_reminders", id: :uuid, default: -> { "uuid_generate_v4()" }, force: :cascade do |t|
    t.uuid "signature_id"
    t.index ["signature_id"], name: "index_verify_reminders_on_signature_id"
  end

  add_foreign_key "verify_reminders", "signatures", on_delete: :cascade
end

```

--------

## Expected behavior

I expect that running rubocop on this file would not report any offense.

## Actual behavior

I get the following report:

```
app/models/signature.rb:11:3: C: Rails/HasManyOrHasOneDependent: Specify a :dependent option.
  has_many :verify_reminders
```

## Steps to reproduce the problem

Run rubocop on a fail containing the code above.

## RuboCop version

Include the output of `rubocop -V`. Here's an example:

```
➔ bundle exec rubocop -v
0.51.0
```

**Comment by pocke** [comment]
You can disable this cop with `.rubocop.yml` or ignore the offenses in your case. 

```yaml
# .rubocop.yml
Rails/HasManyOrHasOneDependent:
  Enabled: false
```

RuboCop cannot know presences of foreign keys. So RuboCop cannot have this feature.
But we can probably add another cop to check foreign key.


```ruby
# db/migrate/2017xxxxxx_some_migration.rb
create_table :users do |t|
  t.belongs_to :repository
# ^^^^^^^^^^^^^^^^^^^^^^^^ Did you forget `foreign_key`?
end
```

**Comment by robbl-as** [comment]
I know that there is no perfect solution to my issue but maybe we could improve the situation. In the given scenario it could be possible to check for the foreign key in the schema.rb file. Though, I have no overview about other possibilities within rubocop.

**Comment by pocke** [comment]
True. But we don't have a cop that reads other files such as `schema.rb`. So I think we cannot implement the feature immediately.
